docs/docs/llms/llm-intent.mdx

```diff
@@ -0,0 +1,162 @@
+---
+id: llm-intent
+sidebar_label: Intent Classification with LLMs
+title: Using LLMs for Intent Classification
+abstract: |
+  Intent classification using Large Language Models (LLM) and
+  a method called retrieval augmented generation (RAG).
+---
+
+import RasaLabsLabel from "@theme/RasaLabsLabel";
+import RasaLabsBanner from "@theme/RasaLabsBanner";
+
+<RasaLabsLabel />
+
+<RasaLabsBanner />
+
+## Key Features
+
+1. **Few shot learning**: The intent classifier can be trained with only a few
+   examples per intent. New intents can be bootstrapped and integrated even if
+   there are only a handful of training examples available.
+2. **Fast Training**: The intent classifier is very quick to train.
+3. **Multilingual**: The intent classifier can be trained on multilingual data
+   and can classify messages in many languages, though performance will vary across LLMs.
+
+## Demo
+
+TODO
+
+## Using the LLM-based Intent Classifier in Your Bot
+
+To use the LLM-based intent classifier in your bot, you need to add the
+`LLMIntentClassifier` to your NLU pipeline in the `config.yml` file.
+
+```yaml-rasa title="config.yml"
+pipeline:
+# - ...
+  - name: rasa_plus.ml.LLMIntentClassifier
+# - ...
+```
+
+The LLM-based intent classifier requires access to an LLM model API. You can use any
+OpenAI model that supports the `/completions` endpoint. 
+We are working on expanding the list of supported
+models and model providers.
+
+## How does the classification work
+
+The intent classification relies on retrieval augmented generation, which is a
+method that combines the benefits of retrieval-based and generation-based
+approaches. The steps are:
+
+1. Embed all intent examples and store their embeddings during the training
+   phase.
+2. During the prediction phase, find the three intent examples from the training
```

Comment:
`find the five intent examples` based on `DEFAULT_NUMBER_OF_INTENT_EXAMPLES = 5`